core/src/main/java/org/infinispan/util/logging/Log.java

```diff
@@ -2367,4 +2368,10 @@ CacheConfigurationException storeConfiguredHasBothReadAndWriteOnly(String storeC
    @LogMessage(level = WARN)
    @Message(value = "Attribute '%s' of element '%s' has been deprecated since schema version %d.%d. Refer to the upgrade guide", id = 688)
    void attributeDeprecated(String name, String element, int major, int minor);
+
+   @LogMessage(level = WARN)
+   @Message(value = "Recovering cache '%s' but there are missing members, known members %s of a total of %s", id = 689)
+   void recoverFromStateMissingMembers(String cacheName, List<Address> members, int total);
+
+   MissingMembersException recoverFromStateMissingMembers(String cacheName, List<Address> members, String total);
```

Comment:
This needs a @Message on it afaik.

Reply:
This is a bit of hack, but I can't find it in the docs now :(
Since this has the same signature as the method above, it will inherit the same message and code. I did this so that the warning and exception have the same code. It is hacky because in one, the `total` is int, and in another is a String.